NEWS.md

```diff
@@ -2,6 +2,8 @@
 
 ## New Features
 
+- New function `derive_vars_cat()` for deriving pairs of variables or more, e.g. 
+`MCRITyML` & `MCRITyMN`. (#2480)
```

Comment:
Can we include `AVALCATy` and `AVALCAyN` here as well?  Actually this is all AVALCAT examples - do want MCRITyML & MCRITyMN?  Is this still helpful for those variables?

Reply:
I could not find the words "MCRITyML" or "MCRITyMN" in the admiral repo. 
Are we deriving them anywhere manually, to your knowledge, @bms63 ?

Reply:
@bms63 would you like me to write an additional example in the function example section following this outline: 
https://chatgpt.com/share/66f251c6-0a44-8011-a0ec-4466d1787bcb

(Of course with the new function...)

Reply:
Ah I just updated the example section. 
You can do this using this function, but not in a very nice way: 

```
# Let's derive both the MCRITyML and the MCRITyMN variables
adlb <- tibble::tribble(
  ~USUBJID,        ~PARAM, ~AVAL, ~AVALU, ~ULN, ~LLN,
  "01-701-1015", "ALT",  150,   "U/L",  40,   10,
  "01-701-1023", "ALT",  70,    "U/L",  40,   10,
  "01-701-1036", "ALT",  130,   "U/L",  40,   10,
  "01-701-1048", "ALT",  30,    "U/L",  40,   10,
  "01-701-1015", "AST",  50,    "U/L",  35,   5
)

# Deriving MCRIT1ML: ALT > 3x ULN
definition_mcrit1ml <- exprs(
  ~PARAM, ~condition, ~MCRIT1ML,
  "ALT", AVAL > 3 * ULN, "Y",
  "ALT", AVAL <= 3 * ULN, "N"
)

# Deriving MCRIT1MN: ALT < LLN
definition_mcrit1mn <- exprs(
  ~PARAM, ~condition, ~MCRIT1MN,
  "ALT", AVAL > ULN, "Y",
  "ALT", AVAL <= ULN, "N"
)

adlb %>%
  derive_vars_cat(
    definition = definition_mcrit1ml,
    by_vars = exprs(PARAM)
  ) %>%
  derive_vars_cat(
    definition = definition_mcrit1mn,
    by_vars = exprs(PARAM)
  )
```

Reply:
I could update the function to make it more flexible, so it could create both at once. 
Or should this be a separate function?
But what would you expect the input to be? @bms63 @bundfussr 
What we could do:
```
definition<- exprs(
  ~PARAM, ~condition, ~MCRIT1MN, MCRIT1ML,
  "ALT", AVAL > ULN,    "Y",                 NULL,
  "ALT", AVAL <= ULN,  "N",                NULL,
  "ALT", AVAL > ULN,    NULL,             "Y",
  "ALT", AVAL <= ULN,  NULL,             "N"
)
```

However, if there are many variables, e.g. agegr1, agegr2, agegr3, region1, region2, etc. this becomes a reeeeally large sparse table.

Alternatively, we could do something like: 
```
definition<- exprs(
  ~PARAM, ~condition, ~TRUE,          ~FALSE,  ~VAR,
  "ALT", AVAL > ULN,    "Y",                 "N",         MCRIT1ML,
  "ALT", AVAL > ULN,    "Y",             "N",             MCRIT1MN,
)
```
But this would not work well for the AVALCAT variables, we would have to allow FALSE to be missing to keep the regular functionality as well...

Reply:
I don't understand the example. `MCRIT1MN` must be numeric. The `MCRITy*` variables should be used for criteria with more than two levels. For binary criteria the `CRITy*` variables should be used.

For an `MCRITy` example I would expect something like
```
definition <- exprs(
  ~condition,                       ~MCRIT1ML, ~MCRIT1MN,
  AVAL <= ANRHI ,                   "<=ULN",           1,
  ANRHI < AVAL & AVAL <= 3 * ANRHI, ">1-3*ULN",        2,
  3 * ANRHI < AVAL,                 ">3*ULN",          3
)
```

Reply:
Ah thats good news, this is easily done with the new function. 
Let me update the example 👍